Few minor improvements for Waterfox Classic

hawkeye116477 opened the pull request:
That should fix some problems which I encountered when building on Windows and restore user-agent overrides for Linux and playing some movies.

hawkeye116477 commented:
@MrAlex94 That should fix YouTube playback on Windows, some problems which I encountered when building on Windows and restore user-agent overrides for Linux.

grahamperrin commented on browser/installer/package-manifest.in:
```diff
@@ -146,7 +146,7 @@
 #endif
 @RESPATH@/browser/blocklist.xml
-#ifdef XP_WIN
+#if defined(XP_WIN) || defined(XP_LINUX)
```
Can this be enhanced to include (edge case) FreeBSD?

A few weeks ago I set aside a note of the likely definition … don't have it handy (sorry) …

grahamperrin replied:
As this (pull request) was not merged before distribution of Waterfox Classic `2019.10 (20191023220600)`, I posted something to Reddit for the AMO use case: 

https://www.reddit.com/comments/d79h2l/-/f5htjhn/

grahamperrin commented:
Re: https://github.com/MrAlex94/Waterfox/commit/fbb4667b0679b6984b418b06bf4fcd9471c41421#r35573351 I'll find an example of an OS-specific add-on …

hawkeye116477 commented:
Ah, conflict and I'll need to do that again.

hawkeye116477 commented:
I updated libnestegg again, but now from good revision :smile:

hawkeye116477 commented:
Small update:
I little changed look of message about restart needed after locale selecting.
![Screenshot_20191022_140726](https://user-images.githubusercontent.com/19818572/67284155-599a3300-f4d5-11e9-9809-d13dcc515ed9.png)
Added also some useful features from Cyberfox: restart button in PanelUI and Menu, copying current tab URL, copying all tab URLs and new advanced preferences.
![Screenshot_20191022_140926](https://user-images.githubusercontent.com/19818572/67284261-9108df80-f4d5-11e9-856c-ddf4755cfeea.png)

I also fixed broken links in about dialog and made some strings translatable.

hawkeye116477 commented:
I added now what @Peacock365 requested, so option for disabling loading images with exceptions, disabling JS with advanced settings and special additions from me => exceptions for JS and option to block notifications permanently with exceptions. 
![Screenshot_20191027_092830](https://user-images.githubusercontent.com/19818572/67631854-2cb29b00-f89c-11e9-9733-da9b9ab55e97.png)
![Screenshot_20191027_093042](https://user-images.githubusercontent.com/19818572/67631879-7602ea80-f89c-11e9-9579-6540f90a496a.png)

Peacock365 commented:
Fantastic work, @hawkeye116477. Have you verified whether or not the new settings also show up when _browser.preferences.useOldOrganization_ is sat to _false_ in Waterfox Classic? Please verify. Also, I hope of course that the new settings will also make it to Waterfox Current at some point. Thumbs up for your work so far!

hawkeye116477 commented:
@Peacock365 No, I haven't verified.

Peacock365 commented:
@hawkeye116477 Not gonna lie, this is a bit of a bummer. I like the new organization of the settings much better, the new organization is also simlar to how the settings in Waterfox Current are organized. I do wonder though, you have replicated these settings in Waterfox Current already: https://github.com/MrAlex94/Waterfox/pull/1213 Is it really that difficult to add them to the new settings page in Waterfox Classic as well? Just curious. I don't t want to press the issue, but the settings being available in one settings page and not in the other is not optimal.

Furthermore, regarding the "Notifications" setting. You may have noticed that _dom.serviceWorkers.enabled_ is set to false in Waterfoc Classic (and in Waterfox Current, too). That was done for good reason, as GMail refused to work in Waterfox Classic with service workers enabled. @grahamperrin, please reference the GMail issue here if possible. Web notifications rely on service workers, though. I'd say the notifications setting should be disabled by default, i.e. it should be opt-in, and if it is checked by the user, the service workers pref should be flipped to true.

hawkeye116477 commented:
`but the settings being available in one settings page and not in the other is not optimal.`
Old Settings are default, also Locale Selector isn't on new settings

hawkeye116477 commented:
`Web notifications rely on service workers, though. `
As I know there are 2 types of notifications => desktop notifications and service worker notifcations.
For example if you go here `https://davidwalsh.name/demo/notifications-api.php`, the notification still work despite that workers are disabled.

grahamperrin commented:
> I like the new organization 

+1

However, #494

grahamperrin commented:
https://davidwalsh.name/demo/notifications-api.php

> how it's possible 

Maybe simplest to think of the **Show a Notification** button as you _pulling_ a notice from the server, and you can not click the button after closing the window. 

Service workers are more commonly associated with _push_ …

Peacock365 commented:
@hawkeye116477 Regarding the "New Tab" settings... When you transfer them to Waterfox Current, please note that the search bar can already be disabled in _about:preferences#home_... The "Enable new tab page", assuming that the new tab page flips to about:blank if the setting gets unticked, is also already present in _about:preferences#home_. Therefore, only the "Disable thumbnail capturing on new tab page" setting would be worthwhile to port to Waterfox Current.

The WebRTC / Geolocation / Referrer Header settings would be great to have in Waterfox Current.

hawkeye116477 commented:
@Peacock365 I know, I saw, I already done that on local, not pushed yet only. I'm also working on adding status bar inspired by Vivaldi :smile:.

grahamperrin commented:
i18n: will everything that's proposed work OOTB, or will parts require translation?

hawkeye116477 commented:
@grahamperrin Yes, if you won't translate then it will be in English, only Polish is translated.
I created experimental project on crowdin => `https://crowdin.com/project/testwater`.
Here is repository with unpacked files => `https://github.com/hawkeye116477/LanguagePacksForWaterfox`.

Peacock365 commented:
But why aren't the new settings translated? This picture shows the picture / Javascript setting translated into Russian, in Cyberfox:

https://www.white-windows.ru/wp-content/uploads/2016/03/4509708_6.png

Also, if the settings are not translated, what is this commit all about?

https://github.com/MrAlex94/Waterfox/pull/1203/commits/b7323f4b3300a12159654435f2a0a99dd9a37ca3

Sorry if the question is silly, I don't get the meaning of the "not translated" statement yet considering the links.

hawkeye116477 commented:
@Peacock365` Also, if the settings are not translated, what is this commit all about?`
I just added English strings.

grahamperrin commented:
**Many** things to be thankful for in this PR. For now (without testing the effects) I'm just watching with interest. 

For now, a couple of points: 

https://github.com/MrAlex94/Waterfox/pull/1203#issuecomment-547819410

> Old settings are default, 

:+1: 

https://github.com/MrAlex94/Waterfox/pull/1203#issuecomment-548707832

> The … Geolocation … settings would be great to have in Waterfox Current.

Do you mean, the settings that must be modified (on some platforms) for location to be determined? 

I mean, the inability of Waterfox to detect location is sometimes a source of confusion; it will be good to have something without dragons at `about:preferences ` 

> in Waterfox Current.

– in Waterfox Classic, yes?

grahamperrin commented:
[browser.restart.purgecache – clear fast restart cache on browser restart – Waterfox Classic](https://www.reddit.com/r/waterfox/comments/e3qc2h/browserrestartpurgecache_clear_fast_restart_cache/)

> … a few questions about fast load data. …
